app/test/e2e/endpoint.spec.js

```diff
@@ -0,0 +1,70 @@
+const nock = require('nock');
+const { TOKENS, microserviceTest } = require('./test.constants');
+const { initHelpers } = require('./utils');
+const { getTestAgent, closeTestAgent } = require('./test-server');
+const Microservice = require('models/microservice.model');
+const Endpoint = require('models/endpoint.model');
+
+const helpers = initHelpers(getTestAgent);
+
+let requester;
+
+const getListEndpoints = () => requester
+    .get('/api/v1/endpoint')
+    .set('Authorization', `Bearer ${TOKENS.ADMIN}`)
+    .send();
+
+const createMicroservice = () => requester
+    .post('/api/v1/microservice')
+    .set('Authorization', `Bearer ${TOKENS.ADMIN}`)
+    .send(microserviceTest);
+
+const openEnpoint = endpoint => requester
+    .get(endpoint)
+    .send();
+
+describe('Endpoints calls', () => {
+    before(async () => {
+        if (process.env.NODE_ENV !== 'test') {
+            throw Error(`Running the test suite with NODE_ENV ${process.env.NODE_ENV} may result in permanent data loss. Please use NODE_ENV=test.`);
+        }
+
+        requester = await getTestAgent();
+        helpers.setRequester(requester);
+        nock.cleanAll();
+    });
+
+    it('Getting a list of endpoints without being authenticated should fail', helpers.isTokenRequired('get', 'plugin'));
+    it('Getting a list of endpoints authenticated not as admin fail', helpers.isAdminOnly('get', 'plugin'));
+    it('Getting a list of endpoints without creating microservice should return empty array', async () => {
+        const response = await getListEndpoints();
+
+        response.status.should.equal(200);
+        response.body.should.instanceof(Array).and.lengthOf(0);
+    });
+    it('Getting a list of endpoints with created microservice should return the result', async () => {
+        await createMicroservice();
```

Comment:
This test is actually testing 2 endpoints: the one that creates a microservice, and the one that lists endpoints. If something fails, it could be one or the other, so it will be harder to debug.

To fix this, instead of creating a microservice using the API, create it using Mongoose models to inject data into the database directly. That way, if this test fails, you'll know that the problem is on the GET endpoint.